plugins/gradle/gradle-plugin/src/main/java/org/wildfly/swarm/plugin/gradle/StartTask.java

```diff
@@ -0,0 +1,354 @@
+            } catch (IOException e) {
+                throw new GradleException("Failed to scan for fractions.", e);
+            }
+
+            // exclude or add further fractions according to user config
+            extension.getFractions().forEach(f -> {
+                if (f.startsWith(EXCLUDE_PREFIX)) {
+                    FractionDescriptor toRemove = FractionDescriptor.fromGav(FractionList.get(), f.substring(1));
+                    getLogger().info("Excluding from detected fractions: {}", toRemove);
+                    detectedFractions.remove(toRemove);
+                } else {
+                    FractionDescriptor toAdd = FractionDescriptor.fromGav(FractionList.get(), f);
+                    getLogger().info("Adding to detected fractions: {}", toAdd);
+                    detectedFractions.add(toAdd);
+                }
+            });
+
+            // combine detected fractions with their dependent fractions
+            final Set<FractionDescriptor> fractionsWithDependencies = new HashSet<>(detectedFractions);
+            fractionsWithDependencies.addAll(detectedFractions.stream()
+                    .flatMap(f -> f.getDependencies().stream())
+                    .collect(Collectors.toSet()));
+
+            getLogger().info("Detected fractions including dependencies:\n  {}",
+                    fractionsWithDependencies.stream()
+                            .map(FractionDescriptor::gav)
+                            .sorted()
+                            .collect(Collectors.joining(INDENTATION_DELIMITER)));
+
+            final Set<ArtifactSpec> fractionsSpecs = fractionsWithDependencies.stream()
+                    .map(ArtifactSpec::fromFractionDescriptor)
+                    .collect(Collectors.toSet());
+
+            // resolve fraction artifacts and include into classpath entries
+            try {
+                List<Path> resolvedPaths = artifactResolvingHelper().resolveAll(fractionsSpecs).stream()
+                        .map(s -> s.file.toPath())
+                        .collect(Collectors.toList());
+                classpathEntries.addAll(resolvedPaths);
+            } catch (Exception e) {
+                throw new GradleException("Failed to resolve fraction dependencies.", e);
+            }
+        }
+
+        getLogger().info("Computed classpath entries:\n  {}",
+                classpathEntries.stream().map(Path::toString).collect(Collectors.joining(INDENTATION_DELIMITER)));
+
+        return classpathEntries;
+    }
+
+    private ArtifactResolvingHelper artifactResolvingHelper() {
+        return new GradleArtifactResolvingHelper(getProject());
+    }
+
+    private void stopProcess(SwarmProcess process) {
+        if (process != null && process.isAlive()) {
+            try {
+                process.stop(10, TimeUnit.SECONDS);
+            } catch (InterruptedException e) {
+                // ignore
+            } finally {
+                process.destroyForcibly();
```

Comment:
I hope there is no side-effect of calling this method on a process that has already exited gracefully. The [Javadocs](https://docs.oracle.com/javase/8/docs/api/java/lang/Process.html#destroyForcibly--) don't provide much insights.

Reply:
I think it just exits when the process was already terminated.